### ISSUE TITLE
Should a ParameterSet conserve original impropers?

### ISSUE THREAD

**Issue by rafwiewiora** [ParmEd/ParmEd#654]
Spinning this out here from my comment here @swails https://github.com/choderalab/openmm/issues/16#issuecomment-203604026 , this is a Parmed issue. 

I noticed this when I was doing debugging by load in leaprc - do dihedrals/improper switching off in `AmberParameterSet`, write out as .dat. 

Parmed does it's own `sorted` on the input atom types in impropers. But in this case this leads to a change in AMBER parameter. It was particularly confusing to me until I realized the difference. 

Should Parmed not to such sorting until it absolutely needs to? Like for `createSystem` it would have to. But should it not do it at `ParameterSet` creation because I expect that what I load in, I write out the same thing? Or more I expect the two files to give me the same results in AMBER. I think that's a reasonable expectation and intuitive.

**Comment by swails** [comment]
Yes, it should preserve order if order matters.  I actually wasn't aware that it did in Amber... :(.

**Comment by swails** [comment]
It doesn't in OpenMM, right?

**Comment by rafwiewiora** [comment]
Hmmm depending on what comes out of this other issue - I'm proposing adding ordering the improper passed to `addTorsion` according to the order in the ffXML as part of the learning what Amber does - so this would introduce the dependence.

**Comment by rafwiewiora** [comment]
Well actually it does matter in OpenMM already. Depending on what you write in the ffXML as the 4th atom, this is currently guaranteed to be the 4th atom in the `addTorsion` call.

**Comment by rafwiewiora** [comment]
The thing is though you're kind of the controller of what OpenMM orders are, because all the AMBER FFs will now have gone through Parmed which enforces a `sorted`!
